### PR TITLE
allows to override storage per attachment

### PR DESCRIPTION
**Pull request by somebody32**
I'll update Readme when initial code review is done

### CODE REVIEW

File: lib/saviour/base_uploader.rb [hunk 1]
```diff
@@ -64,6 +64,10 @@ def store_dir
       @store_dir ||= Uploader::StoreDirExtractor.new(self).store_dir
     end
 
+    def storage
```
Comment (rogercampos) [inline lib/saviour/base_uploader.rb:67]: no need for this no? it's only a class attribute

File: spec/models/base_uploader_spec.rb [hunk 2]
```diff
@@ -44,6 +44,24 @@
...
+    end
+
+    it "allows to override storage" do
+      mocked_storage = double(:mocked_storage)
+      custom_storage = double(:custom_storage)
+      allow(Saviour::Config).to receive(:storage).and_return(mocked_storage)
+
+      subject.store_dir(storage: custom_storage) { "/my/dir" }
+      expect(subject.storage).to eq custom_storage
+      expect(subject.storage).to_not eq mocked_storage
```
Comment (rogercampos) [inline spec/models/base_uploader_spec.rb]: no need for the negative assertion, it's implicit by the first one

**Comment by rogercampos** [comment]
good! I would only add some more high-level tests with a class that have 2 attachments with different storages each, then check crud operations work as expected.

File: lib/saviour/base_uploader.rb (hunk 1 above)
Comment (somebody32) [reply lib/saviour/base_uploader.rb:67]: it is easier from api to call it `uploader.storage` than `uploader.class.storage`, for example in `LifeCycle`

Comment (rogercampos) [reply lib/saviour/base_uploader.rb:67]: well, I see it's used, then I would just remove the memoization here. It's already memoized in the class.

File: README.md [hunk 3]
```diff
@@ -417,9 +419,31 @@ the store dir is a common approach to ensure there will be no collisions. Other
 random token generation.
 
 
+#### with_storage
+
+This method allows you to override storage on per attachment basis. It could be useful if generallly
```
Comment (rogercampos) [inline README.md]: typo

Comment (somebody32) [reply README.md]: arrr!

**Pull request merged**